realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -558,6 +558,41 @@ public void not() {
         }
     }
 
+    @Test
+    public void emptyGroup() {
+        populateTestRealm(); // create TEST_DATA_SIZE objects
+        RealmResults<AllTypes> fives = realm.where(AllTypes.class)
+                .equalTo(AllTypes.FIELD_LONG, 5)
+                .beginGroup()
+                .endGroup()
+                .findAll();
+        // Only on object with value 5
+        assertEquals(1, fives.size());
+    }
+
+    @Test (expected = UnsupportedOperationException.class)
```

Comment:
Generally, we only reserve this exception to things not implemented. A more appropriate exception would be `IllegalStateException`

Reply:
From documentation the various `RealmQuery` methods it seems like invalid queries (syntax errors) are indicated by throwing `UnsupportedOperationException`.